rustler/src/types/primitive.rs

```diff
@@ -60,3 +59,34 @@ impl<'a> Decoder<'a> for bool {
         atom::decode_bool(term)
     }
 }
+
+impl Encoder for f32 {
+    fn encode<'a>(&self, env: Env<'a>) -> Term<'a> {
+        #[allow(clippy::cast_lossless)]
+        unsafe {
+            Term::new(
+                env,
+                rustler_sys::enif_make_double(env.as_c_arg(), *self as f64),
+            )
+        }
+    }
+}
+impl<'a> Decoder<'a> for f32 {
+    fn decode(term: Term) -> NifResult<f32> {
+        #![allow(unused_unsafe)]
+        let mut res: f64 = Default::default();
```

Comment:
We can get rid of the `unsafe` block and call to `enif_get_double` if we decode to `f64` directly:

```rust
let res: f64 = term.decode()?;
```